static/js/message_scroll.js

```diff
@@ -199,6 +200,10 @@ export function initialize() {
             }
             if (event.msg_list.mark_messages_read_enabled()) {
                 unread_ops.notify_server_messages_read(messages, {from: "pointer"});
+            } else if (unread.get_unread_messages(messages).length !== 0 && narrow_state.filter() !== undefined) {
```

Comment:
Mentioned the reason for `narrow_state.filter() !== undefined` check in commit message:
```
The `narrow_state.filter() !== undefined` filter check in
message_scroll.js might seem redundant and actually is
if there wasn't a race condition between the `scroll_timer`
and switching to recent topics. Without that, scrolling a
bit in don't-mark-as-read mode and immediately switching
to recent topics would show the banner because of the
timer leading to execution of 'show banner' code.

If you're wondering why it isn't an issue for other narrows,
the timer gets reset on moving to a different narrow.
```